packages/opentelemetry-plugin-mysql/src/mysql.ts

```diff
@@ -0,0 +1,266 @@
+        );
+      }
+      if (typeof cb === 'function') {
+        cb(...arguments);
+      }
+    };
+  }
+
+  private _patchQuery(connection: mysqlTypes.Connection | mysqlTypes.Pool) {
+    return (originalQuery: Function): mysqlTypes.QueryFunction => {
+      const thisPlugin = this;
+      thisPlugin._logger.debug('MysqlPlugin: patched mysql query');
+
+      return function query(
+        query: string | mysqlTypes.Query | mysqlTypes.QueryOptions,
+        _valuesOrCallback?: unknown[] | mysqlTypes.queryCallback,
+        _callback?: mysqlTypes.queryCallback
+      ) {
+        const spanName = getSpanName(query);
+
+        const span = thisPlugin._tracer.startSpan(spanName, {
+          parent: thisPlugin._tracer.getCurrentSpan() || undefined,
+          kind: SpanKind.CLIENT,
+          attributes: {
+            ...MysqlPlugin.COMMON_ATTRIBUTES,
+            ...getConnectionAttributes(connection.config),
+          },
+        });
+
+        if (typeof query === 'string') {
+          span.setAttribute(AttributeNames.DB_STATEMENT, query);
+        } else if (typeof query === 'object') {
+          if (query.sql) {
+            span.setAttribute(AttributeNames.DB_STATEMENT, query.sql);
+          }
+
+          if (query.values) {
+            span.setAttribute(AttributeNames.MYSQL_VALUES, query.values);
+          }
+        }
+
+        if (arguments.length === 1) {
+          const streamableQuery: mysqlTypes.Query = originalQuery.apply(
+            connection,
+            arguments
+          );
+
+          return streamableQuery
+            .on('error', err =>
+              span.setStatus({
+                code: CanonicalCode.UNKNOWN,
+                message: err.message,
+              })
+            )
+            .on('end', () => {
+              span.end();
+            });
+        }
+
+        if (typeof arguments[1] === 'function') {
+          shimmer.wrap(arguments, 1, thisPlugin._patchCallbackQuery(span));
+        } else if (typeof arguments[2] === 'function') {
+          if (Array.isArray(_valuesOrCallback)) {
+            span.setAttribute(AttributeNames.MYSQL_VALUES, _valuesOrCallback);
+          } else if (arguments[2]) {
+            span.setAttribute(AttributeNames.MYSQL_VALUES, [_valuesOrCallback]);
+          }
+          shimmer.wrap(arguments, 2, thisPlugin._patchCallbackQuery(span));
+        }
+
+        return originalQuery.apply(connection, arguments);
+      };
+    };
+  }
+
+  private _patchCallbackQuery(span: Span): (original: any) => any {
+    return (originalCallback: mysqlTypes.queryCallback) => {
+      return function(
+        err: mysqlTypes.MysqlError | null,
+        results?: any,
+        fields?: mysqlTypes.FieldInfo[]
+      ) {
+        if (err) {
+          span.setStatus({
+            code: CanonicalCode.UNKNOWN,
+            message: err.message,
+          });
+        }
+        span.end();
+        originalCallback(err, results, fields);
```

Comment:
I think it would be safer to the `originalCallback.apply(this, arguments)` to avoid missing addional args (even if there no more args in the current implementation, it would save us some time later if it changes)

Reply:
Since this is the callback passed in by the user, we don't want to override their `this` binding by using `Function.prototype.apply`. What would you think of 
```suggestion
        return originalCallback(...arguments);
```

Reply:
I applied this in https://github.com/dynatrace-oss-contrib/opentelemetry-js/commit/031e60b95b52cb1aaa7628c34f93f782bd44d5d9

PTAL